utilities/write_batch_with_index/write_batch_with_index_test.cc

```diff
@@ -2496,6 +2548,218 @@ TEST_P(WriteBatchWithIndexTest, IndexNoTs) {
   }
 }
 
+TEST_P(WriteBatchWithIndexTest, WideColumnsBatchOnly) {
+  // Tests for the case when there's no need to consult the underlying DB during
+  // queries, i.e. when all queries can be answered using the write batch only.
+
+  ASSERT_OK(OpenDB());
+
+  constexpr size_t num_keys = 6;
+
+  constexpr char delete_key[] = "d";
+  constexpr char delete_merge_key[] = "dm";
+  constexpr char put_entity_key[] = "e";
+  constexpr char put_entity_merge_key[] = "em";
+  constexpr char put_key[] = "p";
+  constexpr char put_merge_key[] = "pm";
+
+  AddToBatch(db_->DefaultColumnFamily(), delete_key);
+  AddToBatch(db_->DefaultColumnFamily(), delete_merge_key);
+  AddToBatch(db_->DefaultColumnFamily(), put_entity_key);
+  AddToBatch(db_->DefaultColumnFamily(), put_entity_merge_key);
+  AddToBatch(db_->DefaultColumnFamily(), put_key);
+  AddToBatch(db_->DefaultColumnFamily(), put_merge_key);
+
+  std::array<Slice, num_keys> keys{{delete_key, delete_merge_key,
+                                    put_entity_key, put_entity_merge_key,
+                                    put_key, put_merge_key}};
+
+  std::array<WideColumns, num_keys> expected{
+      {{},
+       {{kDefaultWideColumnName, "dm1"}},
+       {{kDefaultWideColumnName, "e0"}, {"e", "0"}},
+       {{kDefaultWideColumnName, "em0,em1"}, {"em", "0"}},
+       {{kDefaultWideColumnName, "p0"}},
+       {{kDefaultWideColumnName, "pm0,pm1"}}}};
+
+  // GetFromBatchAndDB
+  {
+    PinnableSlice value;
+    ASSERT_TRUE(batch_->GetFromBatchAndDB(db_, read_opts_, delete_key, &value)
+                    .IsNotFound());
+  }
+
+  for (size_t i = 1; i < num_keys; ++i) {
+    PinnableSlice value;
+    ASSERT_OK(batch_->GetFromBatchAndDB(db_, read_opts_, keys[i], &value));
+    ASSERT_EQ(value, expected[i].front().value());
+  }
+
+  // MultiGetFromBatchAndDB
+  {
+    std::array<PinnableSlice, num_keys> values;
+    std::array<Status, num_keys> statuses;
+    constexpr bool sorted_input = false;
+
+    batch_->MultiGetFromBatchAndDB(db_, read_opts_, db_->DefaultColumnFamily(),
+                                   num_keys, keys.data(), values.data(),
+                                   statuses.data(), sorted_input);
+
+    ASSERT_TRUE(statuses[0].IsNotFound());
+
+    for (size_t i = 1; i < num_keys; ++i) {
+      ASSERT_OK(statuses[i]);
+      ASSERT_EQ(values[i], expected[i].front().value());
+    }
+  }
+
```

Comment:
I guess test cases for `GetEntityFromBatchAndDB` `MultiGetEntityFromBatchAndDB` will be added somewhere around here in the later PRs. nitpick, but TODO would be good to match other TODO comments in `write_batch_with_index.h` :)